### DISCUSSION TITLE
How can I structure my blog posts in a separate `blog` folder?

### DISCUSSION THREAD

**Discussion by nikolailehbrink** [remix-run/remix#10038] [Q&A] [answered]
I am working on a blog using Remix and MDX, inspired by [this repository](https://github.com/pcattori/remix-blog-mdx/tree/main/app/routes). I want to achieve better separation between my blog posts and the rest of the app. My goal is to place all blog `.mdx` files in a dedicated `blog` folder and have the corresponding URLs structured like `/blog/post-name`. Here’s how I would like the structure to look:

```text
app/
  routes/
    blog/
      introduction.mdx → `/blog/introduction`
      another-post.mdx → `/blog/another-post`
```

I looked into using the [remix-flat-routes](https://github.com/kiliman/remix-flat-routes) package to achieve this clean structure but couldn’t figure out how to do it. Does anyone have a suggestion or know if this package can support my use case? According to the [remix-routing demo](https://remix-routing-demo.netlify.app/and/this/works/too), this could be achieved in v1.

Thanks in advance for any insights!

**Comment by zxlvera** [comment]
I had a similar issue and resolved using `import.meta.glob('../blogs/*.mdx');`

https://github.com/orgs/mdx-js/discussions/2542

**Reply by nikolailehbrink** [reply]
Thanks for the answer @zxlvera. My problem was just the creation of the `blog` folder and a file route inside that, which can be done by suffixing the `+`  to `blog+` like @kiliman mentioned in [his answer](https://github.com/remix-run/remix/discussions/10038#discussioncomment-10811761) 🙂

**Comment by kiliman** [comment, marked as answer]
For nested routing with `remix-flat-routes`, just name the folder to `blog+`, otherwise you'd need to update every file with `blog.` prefix (e.g, `blog.introduction.mdx`)

**Reply by nikolailehbrink** [reply]
Thanks @kiliman! Exactly, what I was looking for. Read the docs before, but seems like not attentive enough!
